### PR TITLE
feat: add "UpdateSchema" operation to class "DataFrameOperation"

### PR DESCRIPTION
**Pull request by graczhual**

### CODE REVIEW

**Comment by GravitiRob** [comment]
Task linked: [CU-2h0z80y 实现支持追加列的 Operation](https://app.clickup.com/t/2h0z80y)

File: graviti/dataframe/frame.py [hunk 1]
```diff
@@ -154,9 +154,8 @@ def __setitem__(self, key: str, value: Union[Iterable[Any], _C]) -> None:
         self._columns[key] = column
         if key not in self._column_names:
             self._column_names.append(key)
-        # TODO: Need add corresponding operations.
         if self.operations is not None:
-            self.operations.append(UpdateData(self[[key]]))
+            self.operations.extend((UpdateSchema(self.schema), UpdateData(self[[key]])))
```
Comment (linjiX) [inline graviti/dataframe/frame.py:158]: will append twice faster?

File: graviti/operation/frame.py [hunk 2]
```diff
@@ -113,6 +118,57 @@ def do(  # pylint: disable=invalid-name
...
+            dataset: Name of the dataset, unique for a user.
+            draft_number: The draft number.
+            sheet: The sheet name.
+            jobs: The number of the max workers in multi-thread operation.
+
+        """
+        schema = self.schema.to_yaml()
+        pyarrow_schema = pa.schema(self.schema.fields.to_pyarrow())  # type: ignore[attr-defined]
+        arrow_schema = base64.encodebytes(pyarrow_schema.serialize().to_pybytes()).decode("ascii")
+        avro_schema = json.dumps(convert_portex_schema_to_avro(self.schema))
```
Comment (Lee-000) [inline graviti/operation/frame.py:158]: Does this part of logic exist in other operations as well? Can we reuse them?

Comment (graczhual) [reply graviti/operation/frame.py:158]: yes, CreateSheet operation also use this part. should we add a method to schema or add a function to sheet operation or frame operation

File: graviti/dataframe/frame.py (hunk 1 above)
Comment (graczhual) [reply graviti/dataframe/frame.py:158]: extend tuple will be a little faster than append twice

**Review by linjiX — APPROVED** [review]

**Review by Lee-000 — APPROVED** [review]

**Pull request merged**